### PR TITLE
External view contract - part 1

### PR DESCRIPTION
**Pull request by andrei-marinica**
Nothing useful can be produced yet with the functionality here, but this is the foundation for having endpoints end up in multiple contracts.

The `meta` crates now have support for splitting the main contract into several.

The `wasm` crate still needs to be present, but the other secondary wasm crates will be generated based on the initial one.

Also added `--wasm-suffix` flag to the build command, which will add the suffix to all generated contracts.

### CODE REVIEW

File: elrond-wasm-debug/src/meta/meta_main.rs [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+use std::env;
+
+use super::meta_config::MetaConfig;
+
+pub fn perform<AbiObj: ContractAbiProvider>() {
+    let original_contract_abi = <AbiObj as ContractAbiProvider>::abi();
+    super::meta_validate_abi::validate_abi(&original_contract_abi).unwrap();
+
+    let args: Vec<String> = env::args().collect();
+    let meta_config = MetaConfig::create(&original_contract_abi, args.as_slice());
```
Comment (dorin-iancu) [inline elrond-wasm-debug/src/meta/meta_main.rs:11]: Seems to me like the `perform` function uses the second argument (first was the current executable path iirc), while the `MetaConfig::create` function only uses the ones after, so I suggest only passing `args.as_slice()[2..]` or similar.

**Review by dorin-iancu — APPROVED** [review]

**Review by alyn509 — APPROVED** [review]

File: elrond-wasm-debug/src/meta/meta_config.rs [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            return wasm_name_override.clone();
+        }
+        if let Some(wasm_suffix) = &self.wasm_name_suffix {
+            format!(
+                "{}-{}.wasm",
+                contract_metadata.output_base_name, wasm_suffix
+            )
+        } else {
+            contract_metadata.wasm_output_name()
+        }
```
[the remark this answers is not in the document]
Comment (andrei-marinica) [reply elrond-wasm-debug/src/meta/meta_config.rs:24]: For `Option` I noticed Rust was suggesting if let - else at some point, so I got used to using it.

File: elrond-wasm-debug/src/meta/meta_config.rs [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+                    .expect("argument `--wasm-name` must be followed by the desired name");
+                result.wasm_name_override = Some(name.clone());
+            },
+            "--wasm-suffix" => {
+                let suffix = iter
+                    .next()
+                    .expect("argument `--wasm-suffix` must be followed by the desired suffix");
+                result.wasm_name_suffix = Some(suffix.clone());
+            },
+            _ => {},
```
[the remark this answers is not in the document]
Comment (andrei-marinica) [reply elrond-wasm-debug/src/meta/meta_config.rs:85]: It's supposed to be open-ended, there are more attributes that cam come from erdpy, which should be accepted

File: elrond-wasm-debug/src/meta/meta_main.rs (hunk 1 above)
Comment (andrei-marinica) [reply elrond-wasm-debug/src/meta/meta_main.rs:11]: We are parsing them anyway, one by one, so some extra arguments slipping through are not an issue. Somehow I thought it's more flexible if it has access to all of them. Works both ways though

**Pull request merged**